### PR TITLE
feat: manage health settings using Coder API

### PR DESCRIPTION
**Pull request by mtojek**
Related: https://github.com/coder/coder/issues/10712

This PR exposes new API endpoints to modify health page settings.

Follow-up:
- Expose `dismissed` property via `/debug/health`

### CODE REVIEW

File: coderd/debug_test.go [hunk 1]
```diff
@@ -232,6 +233,52 @@ func TestDebugHealth(t *testing.T) {
 	})
 }
 
+func TestHealthSettings(t *testing.T) {
```
Comment (johnstcn) [inline coderd/debug_test.go:236]: 1) Can we add a test case where we un-dismiss?
2) This should be audited, so we want to add an auditor and assert that an audit action took place.

File: codersdk/health.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	res, err := c.Request(ctx, http.MethodPut, "/api/v2/debug/health/settings", settings)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+	if res.StatusCode != http.StatusOK {
+		return ReadBodyAsError(res)
+	}
+	return nil
+}
```
Comment (johnstcn) [inline codersdk/health.go:40]: nit: `PutHealthSettings`?

File: coderd/debug.go [hunk 3]
```diff
@@ -107,6 +112,104 @@ func formatHealthcheck(ctx context.Context, rw http.ResponseWriter, r *http.Requ
...
+	settingsJSON, err := json.Marshal(&settings)
+	if err != nil {
+		httpapi.Write(r.Context(), rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Failed to marshal health settings.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	err = api.Database.UpsertHealthSettings(ctx, string(settingsJSON))
```
Comment (johnstcn) [inline coderd/debug.go:191]: nit: For PUT idempotency, should we check if existing == to be updated and return 304 if so?

**Review by johnstcn — COMMENTED** [review]
LGTM, just a couple things I'd like to see added first.

File: coderd/debug.go (hunk 3 above)
Comment (mtojek) [reply coderd/debug.go:191]: Yeah, that's why I was thinking about POST too :) I can adapt this logic 👍

File: coderd/debug_test.go (hunk 1 above)
Comment (johnstcn) [reply coderd/debug_test.go:236]: Something like this perhaps?

https://github.com/coder/coder/blob/main/coderd/workspaces_test.go#L597-L612

https://github.com/coder/coder/blob/main/coderd/workspaces.go#L313-L321

File: codersdk/health.go (hunk 2 above)
Comment (mtojek) [reply codersdk/health.go]: Done

File: coderd/debug.go (hunk 3 above)
Comment (mtojek) [reply coderd/debug.go:206]: Fixed

File: coderd/debug_test.go (hunk 1 above)
Comment (mtojek) [reply coderd/debug_test.go:236]: 1. Undismiss test case done

Comment (mtojek) [reply coderd/debug_test.go:236]: 2. I added support for auditing, but I have no idea if this is sufficient. Keep in mind that "health settings" are stored as site configs, so in this case, the resource (and its ID) is artificial.

Comment (johnstcn) [reply coderd/debug_test.go:236]: You can add a mock auditor into the unit tests and assert that the changes were audited as expected.

**Review by johnstcn — APPROVED** [review]

**Pull request merged**